src/check.rs

```diff
@@ -65,13 +88,59 @@ fn check_file_url(url: &Url, _ctx: &CheckContext) -> Result<(), CheckError> {
 
     if path.is_file() || path.join("index.html").is_file() {
         debug!("Linked file at path {} does exist.", path.display());
-        Ok(())
+        match url.fragment() {
+            Some(fragment) => check_file_fragment(&path, fragment),
+            None => Ok(()),
+        }
     } else {
         debug!("Linked file at path {} does not exist!", path.display());
         Err(CheckError::File(path))
     }
 }
 
+fn check_file_fragment(path: &Path, fragment: &str) -> Result<(), CheckError> {
+    debug!("Checking fragment {} of file {}.", fragment, path.display());
+    let (_, targets) = parse_html_file(path, path);
```

Comment:
Is there any reason to parse this twice, and in a separate function from `unavailable_urls`? I think that would also get you HTTP support for free.

Reply:
I don't think I've explained the logic well enough or that the code speaks of it good enough, it is a quick/first implementation. Let me try to explain this better here.

When checking a link containing a fragment, say `b.html#fragment` in a anchor tag in file `a.html`, the algorithm needs to look in `b.html` and see if a node with the `id="fragment"` exists there. The first, naive, implementation is what I've submitted here: parse the linked file (`b.html`) and see what targets exist there and check if the wanted fragment is one of them. This is similar approach to the one already taken currently to check if a linked file exists (using `Path::is_file()`). The call to `Path::is_file()` could be invoked several times for the same path (consider `a.html` having multiple links to `b.html`). Having this in mind I've implemented the memoization, so that the performance penalty of parsing the same file multiple times is not paid in full.

I can think of a better but more complex approach, whereby while parsing any file a data model containing links and fragments is generated and kept in memory. The complexity of that approach is that since the link checking is done in parallel over several threads the core of deadlinks would need to change to implement queue-like/asynchronous algorithm. And that seemed too big of a change for a first pass at this functionality.